src/core/geometry/qgsmultilinestring.cpp

```diff
@@ -178,3 +179,39 @@ bool QgsMultiLineString::wktOmitChildType() const
   return true;
 }
 
+QgsMultiLineString *QgsMultiLineString::measuredLine( double start, double end ) const
+{
+
+
+  QgsMultiLineString *result{createEmptyWithSameType()};
+
+  if ( !result->convertTo( QgsWkbTypes::addM( mWkbType ) ) )
+  {
+    return result;
+  }
+
+  if ( isEmpty() )
+  {
+    return result;
+  }
+
+  /* Calculate the total length of the line */
+  const double length{this->length()};
+  const double range{end - start};
+  double lengthSoFar{0.0};
+
+  result->reserve( numGeometries() );
+  for ( int i = 0; i < numGeometries(); i++ )
+  {
+    const double subLength{geometryN( i )->length()};
+
+    const double subStart{ ( start + range *lengthSoFar / length ) };
+    const double subEnd{ ( start + range * ( lengthSoFar + subLength ) / length ) };
```

Comment:
Need some logic to handle length == 0 here to avoid divide by 0

Reply:
Will returns NaN as for PostGIS.

`SELECT ST_AsText(ST_AddMeasure('multilinestring(( 0 0, 0 0, 0 0), (1 1, 1 1, 1 1) )', 10 , 20)), ST_Length('multilinestring(( 0 0, 0 0, 0 0), (1 1, 1 1, 1 1) )')`

=> 
`MULTILINESTRING M ((0 0 NaN,0 0 NaN,0 0 NaN),(1 1 NaN,1 1 NaN,1 1 NaN)) | 0.0`

I added a test case